### ISSUE TITLE
test_store_resource_deprecated failure

### ISSUE THREAD

**Issue by mdickinson** [enthought/traits#1677]
When running tests on the current main branch, in an environment that doesn't have `setuptools` installed, I'm seeing the following failure:

```python
======================================================================
ERROR: test_store_resource_deprecated (traits.util.tests.test_resource.TestResource)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/mdickinson/Enthought/ETS/traits/traits/util/tests/test_resource.py", line 31, in test_store_resource_deprecated
    store_resource(
  File "/Users/mdickinson/Enthought/ETS/traits/traits/util/resource.py", line 200, in store_resource
    raise RuntimeError(
RuntimeError: Resource not found for project "traits": traits/__init__.py
```

We may need to simply skip this test if `setuptools` is not installed.

**Comment by mdickinson** [comment]
Hmm; it's a bit more than just the test failing: `find_resource` itself is broken if `setuptools` is not installed. However, since it's deprecated and we're intending to remove it, it's likely not worth fixing. The purpose of the test is to check the deprecation, not the correct functioning of `store_resource`, so I think it's fine to only run the test when `setuptools` is installed.

**Comment by mdickinson** [comment]
One additional note that might explain why the CI runs weren't picking this test failure up: the test failure only appears for editable installs (installed with `pip install -e .`). If I install instead with `pip install .`, I don't see the test failure.

**Comment by mdickinson** [comment]
Fixed in #1679.
